trainers/base_trainer.py

```diff
@@ -197,10 +255,10 @@ def run_profile(self):
         ) as prof:
             for i in range(10):
                 if i <= 3:
-                    self._run_step()
+                    self._run_step(i) ## set the 'epoch' to ensure shuffle
```

Comment:
btw is this still necessary?

Reply:
Yes. I implemented it without it previously and I printed the X and y. They kept returning the same sets over each iteration. In other words, gradient_accumulation_step 0 returns the same across all iterations. That led to a smooth loss value. 

Revisiting the Pytorch DDP implementation and following this [thread](https://discuss.pytorch.org/t/why-is-sampler-set-epoch-epoch-needed-for-distributedsampler/149672/2), setting the epoch will help to shuffle the data. This means the gradient_accumulation_step 0 for iteration 0 and iteration 1 will no longer be the same.